### ISSUE TITLE
Bug encountered while running fibonacci example

### ISSUE THREAD

**Issue by alanpjohn** [ARgorithm/Toolkit#6]
### Why is it null ? 

Encountered during the cli test run with server 
I ran the fibonacci code with default parameters
and this was the output
```json
{
  "data": [
    {
      "comments": "creating the stack",
      "state_def": {
        "body": [],
        "variable_name": "stack"
      },
      "state_type": "stack_declare"
    },
    {
      "comments": "",
      "state_def": {
        "body": [
          0
        ],
        "element": 0,
        "variable_name": "stack"
      },
      "state_type": "stack_push"
    },
    {
      "comments": "",
      "state_def": {
        "body": [
          0,
          1
        ],
        "element": 1,
        "variable_name": "stack"
      },
      "state_type": "stack_push"
    },
    {
      "comments": "this will store the answer",
      "state_def": {
        "value": null,
        "variable_name": "answer"
      },
      "state_type": "variable_declare"
    },
    {
      "comments": "this will store the second highest",
      "state_def": {
        "value": null,
        "variable_name": "answer"
      },
      "state_type": "variable_declare"
    }
  ],
  "status": "run_default"
}
```

the code of the fibonacci doesnt look good either tbh
it could be a error in the variable class or could be an error in the fibonacci code

```python
import ARgorithmToolkit

def run(**kwargs):
    n = kwargs["n"]
    algo = ARgorithmToolkit.StateSet()
    st = ARgorithmToolkit.Stack("stack" , algo , "creating the stack")
    st.push(0)
    st.push(1)    
    curr = ARgorithmToolkit.Variable("answer",algo , comments="this will store the answer")
    temp = ARgorithmToolkit.Variable("answer",algo , comments="this will store the second highest")
    if n > 0:
        for i in range(1,n):
            temp.value =  st.pop()
            curr.value = temp.value + st.top()
            st.push(temp.value)
            st.push(curr.value)
    return algo
```

you can replicate these results
```bash
python -m ARgorithmToolkit test
```
`ARgorithmToolkit` : `0.0.6`

**Comment by alanpjohn** [comment]
@yatharthmathur maybe we need to increase the quality of testing
currently testing is only being done for the state_type
i think it would be better to have more in depth tests written

**Comment by yatharthmathur** [comment]
I'll add more stringent test cases for other structures too tomorrow.

**Comment by alanpjohn** [comment]
fixed in #7
